src/ethereum_spec_tools/new_fork.py

```diff
@@ -0,0 +1,169 @@
+"""
+
+parser = argparse.ArgumentParser(
+    description=DESCRIPTION,
+    formatter_class=argparse.RawDescriptionHelpFormatter,
+)
+
+parser.add_argument("--from_fork", dest="from_fork", type=str, required=True)
+parser.add_argument("--to_fork", dest="to_fork", type=str, required=True)
+parser.add_argument("--from_test", dest="from_test", type=str)
+parser.add_argument("--to_test", dest="to_test", type=str)
+
+
+def find_replace(dir: str, find: str, replace: str, file_pattern: str) -> None:
+    """
+    Replace the occurrance of a certain text in files with a new text
+    """
+    for path, _, files in os.walk(dir):
+        for filename in fnmatch.filter(files, file_pattern):
+            file_path = os.path.join(path, filename)
+            with open(file_path) as f:
+                s = f.read()
+            s = s.replace(find, replace)
+            with open(file_path, "w") as f:
+                f.write(s)
+
+
+class ForkCreator:
+    """
+    Object to create base code for a new fork from the previous fork.
+    """
+
+    def __init__(
+        self,
+        from_fork: str,
+        to_fork: str,
+        from_test_names: str,
+        to_test_names: str,
+    ):
+        self.script_file = os.path.abspath(__file__)
+        # Assumes that the root folder is 3 folders up
+        self.root_folder = os.path.dirname(
+            os.path.dirname(os.path.dirname(self.script_file))
+        )
+        self.package_folder = os.path.join(self.root_folder, "src/ethereum")
+        self.optimized_package_folder = os.path.join(
+            self.root_folder, "src/ethereum_optimized"
+        )
+        self.test_folder = os.path.join(self.root_folder, "tests")
+
+        # Get the fork specific data for from fork
+        self.from_fork = from_fork
+        self.from_package = self.from_fork.replace(" ", "_").lower()
+        self.from_path = os.path.join(self.package_folder, self.from_package)
+        self.from_optimized_path = os.path.join(
+            self.optimized_package_folder, self.from_package
+        )
+        self.from_test_path = os.path.join(self.test_folder, self.from_package)
+
+        # Get the fork specific data for to fork
+        self.to_fork = to_fork
+        self.to_package = self.to_fork.replace(" ", "_").lower()
+        self.to_path = os.path.join(self.package_folder, self.to_package)
+        self.to_optimized_path = os.path.join(
+            self.optimized_package_folder, self.to_package
+        )
+        self.to_test_path = os.path.join(self.test_folder, self.to_package)
+
+        self.from_test_names = from_test_names
+        self.to_test_names = to_test_names
+
+    def duplicate_fork(self) -> None:
+        """
+        Copy the relevant files/folders from the old fork
+        """
+        os.system(f"cp -R {self.from_path} {self.to_path}")
+        os.system(f"cp -R {self.from_optimized_path} {self.to_optimized_path}")
+        os.system(f"cp -R {self.from_test_path} {self.to_test_path}")
```

Comment:
Would something like [`shutil.copytree`](https://docs.python.org/3/library/shutil.html#shutil.copytree) work here? I doubt `cp` works the same on Windows :(

Reply:
Oh, there's also path-with-spaces issues when you use `os.system` if I remember correctly.